src/main/java/org/opentripplanner/routing/api/request/framework/Units.java

```diff
@@ -0,0 +1,147 @@
+package org.opentripplanner.routing.api.request.framework;
+
+import org.opentripplanner.util.lang.DoubleUtils;
+
+/**
+ * This utility can be used to perform sanity checks on common number types. It will also normalize
+ * the numbers. Supported types are:
+ * <ul>
+ *   <li>cost - A generalized-cost</li>
+ *   <li>reluctance - Reluctance or factor.</li>
+ *   <li>slack - Time/slack/duration. Unit: seconds (s)</li>
+ *   <li>speed - Unit: meters per second (m/s)</li>
+ *   <li>acceleration - Unit: meters per second squared (m/s^2)</li>
+ *   <li>ratio - Unit: meters per second squared (m/s^2)</li>
+ * </ul>
+ */
+public class Units {
+
+  private static final double ONE_MACH = 340.0;
+
+  /** This is a utility class, it is never instantiated; Hence, private constructor. */
+  private Units() {}
+
+  /**
+   * A generalized-cost. Zero(0) or positive integer.
+   * <p>
+   * Unit: Human cost of riding transit for 1 second ($)
+   */
+  public static int cost(int value) {
+    if (value < 0) {
+      throw new IllegalArgumentException("Negative generalized-cost not expected: " + value);
+    }
+    return value;
+  }
+
+  /**
+   * Reluctance of factor from zero(0) to positive infinitive.
+   * Number of decimals used are: 2 for values less than 2.0, 1 for values less than 10.0, and
+   * zero for values above 10.0.
+   * <p>
+   * Unit: meters per second (scalar)
+   */
+  public static double reluctance(double value) {
+    if (value < 0.0) {
+      throw new IllegalArgumentException("Negative reluctance not expected: " + value);
+    }
+    if (value < 2.0) {
+      return DoubleUtils.roundTo2Decimals(value);
+    }
+    if (value < 10.0) {
+      return DoubleUtils.roundTo1Decimal(value);
+    }
+    return DoubleUtils.roundToZeroDecimals(value);
+  }
+
+  /**
+   * Time/slack/duration - A constant amount of time.
+   */
+  public static int slack(int seconds) {
```

Comment:
Should we rename this to duration? A slack is a duration, but eg. a parkTime is not a slack?

Reply:
I guess that depends on the definition of slack. From the system/algorithm point of view it is - it gives you some slack to complete the task - it is not the actual time it takes. But from the users point of with it it is the duration/time requested to park. I think "duration" is a bit vague and broad. I am happy to add another method as well, but time is not correct, so I am unsure what to call it. I did think trough it, but its hard.

Reply:
I can do both `slack` and `duration` maybe?

Reply:
There was only one slack, so I changed it to duration. These values should be changed from `int` to a type-safe type, but not in this PR.